### PR TITLE
chore(dashpay): merge master [NMA-1439]

### PR DESCRIPTION
**Pull request by Syn-McJ**
Merging 8.1.1 into DashPay

## Issue being fixed or feature implemented
- `BottomNavigationView` was realigned with the master branch, and `ContactsFragment` navigation was moved to the navigation controller to align with the new pattern.
- The new `ReceiveInfoView` was modified to add support for username and avatar. Some avatar-specific code was moved to `common` and cleaned up from dependency on `dpp`.

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: .github/workflows/dashpay.yml [hunk 1]
```diff
@@ -68,12 +68,21 @@ jobs:
...
+        echo "$SERVICE_PROPERTIES" > service.properties
+        echo "$LOCAL_PROPERTIES" > local.properties
+        echo "$APP_DISTRIBUTION_KEY_JSON" > .deploy/app-distribution-key.json
+        echo "$GC_STORAGE_SERVICE_ACCOUNT_KEY_JSON" > .deploy/gc-storage-service-account.json
+      env:
+        GOOGLE_SERVICES_JSON : ${{secrets.GOOGLE_SERVICES_JSON}}
+        SERVICE_PROPERTIES: ${{secrets.SERVICE_PROPERTIES}}
+        LOCAL_PROPERTIES: ${{secrets.LOCAL_PROPERTIES}}
+        APP_DISTRIBUTION_KEY_JSON: ${{secrets.APP_DISTRIBUTION_KEY_JSON}}
+        GC_STORAGE_SERVICE_ACCOUNT_KEY_JSON: ${{secrets.GC_STORAGE_SERVICE_ACCOUNT_KEY_JSON}}
```
Comment (Syn-McJ) [inline .github/workflows/dashpay.yml:85]: This was a later change, but we can include it to keep CI running.

File: wallet/src/de/schildbach/wallet/ui/dashpay/ContactsFragment.kt [hunk 2]
```diff
@@ -52,56 +60,66 @@ import org.dash.wallet.common.ui.dialogs.AdaptiveDialog
...
+
+enum class ContactsScreenMode {
+    SEARCH_CONTACTS,
+    SELECT_CONTACT,
+    VIEW_REQUESTS
+}
 
 @AndroidEntryPoint
-class ContactsFragment : BottomNavFragment(R.layout.fragment_contacts_root),
+class ContactsFragment : Fragment(),
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/dashpay/ContactsFragment.kt:72]: `ContactsFragment` was added to the navigation graph and related changes were made.

File: wallet/src/de/schildbach/wallet/ui/dashpay/utils/ProfilePictureDisplay.kt [hunk 3]
```diff
@@ -1,135 +1,43 @@
...
+        display(
+            avatarView,
+            dashPayProfile.avatarUrl,
+            dashPayProfile.avatarHash,
+            dashPayProfile.username,
+            disableTransition,
+            listener
+        )
+    } else if (hideIfProfileNull) {
+        avatarView.visibility = View.GONE
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/dashpay/utils/ProfilePictureDisplay.kt:41]: Most of `ProfilePictureDisplay` is moved to common and only parts that depend on `DashPayProfile` and `dpp` are left in the wallet as extensions.

File: common/src/main/java/org/dash/wallet/common/ui/receive/ReceiveInfoView.kt [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                binding.usernameLabel.text = context.getString(R.string.username)
+            }
+
+            if (!avatar.isNullOrEmpty()) {
+                ProfilePictureDisplay.display(binding.avatar, avatar, avatarHash, username, false, null)
+            }
+        } else {
+            binding.usernamePreviewPane.isVisible = false
+        }
+    }
```
Comment (Syn-McJ) [inline common/src/main/java/org/dash/wallet/common/ui/receive/ReceiveInfoView.kt:125]: The old `ReceiveInfoView` was removed, new one was extended with username and avatar.

File: wallet/src/de/schildbach/wallet/ui/explore/ExploreFragment.kt [hunk 5]
```diff
@@ -21,64 +21,62 @@ import android.content.Intent
...
             handleStakingNavigation()
         }
 
-        binding.faucetBtn.setOnClickListener {
-            val intent = ExploreActivity.createIntent(requireContext(), ExploreTopic.Faucet)
-            startActivity(intent)
-        }
-
-        setAPY(0.0) // hide the APY
-
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/explore/ExploreFragment.kt:81]: This was removed, but we need the Faucet button functionality.

**Review by HashEngineering — CHANGES_REQUESTED** [review]

File: wallet/src/de/schildbach/wallet/ui/explore/ExploreFragment.kt (hunk 5 above)
Comment (Syn-McJ) [reply wallet/src/de/schildbach/wallet/ui/explore/ExploreFragment.kt]: Good catch, fixed

**Review by HashEngineering — APPROVED** [review]
LGTM

**Pull request merged**